edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Constants.cs

```diff
@@ -71,6 +71,8 @@ public static class Constants
 
         public const string EdgeletApiVersionVariableName = "IOTEDGE_APIVERSION";
 
+        public const string EdgeletProductInfoVariableName = "IOTEDGE_PRODUCTINFO";
```

Comment:
Reiterating what we discussed previously - 
My understanding is that it is possible to update the OS with a new version without destroying the containers on the device. In that case, if we pass the ProductInfo as a Env var, then it will not be updated after the OS update, so a pull based mechanism would be better. Maybe something like adding it to the SystemInfo call on the management socket might fit the bill? https://github.com/Azure/iotedge/blob/master/edgelet/api/managementVersion_2021_05_11.yaml#L426